File: backend-node-portifolio/src/api/data/interface/IGenerics.ts
export interface IGenerics<T>{
    Update(Entity:T):Promise<string>;
    ListAll():Promise<T[]>;
    Save(Entity:T):Promise<string>;
    GetbyId(Id:number): Promise<T>;
    Delete(Entity:T): Promise<string>;
    DeleteById(Id:number): Promise<string>;
}

File: backend/CommentService/src/CommentService.Domain/Interfaces/IGenerics.cs
using System.Linq.Expressions;
namespace CommentService.Domain.Interfaces
{
    public interface IGenerics<T> where T : class
    {
        Task Save(T entity);
        Task Update(T entity);
        Task Delete(T entity);
        Task<T> FindBy(Expression<Func<T, bool>> predicate);
        Task<T> GetById(Guid Id);
        Task<List<T>> List();
        Task<List<T>> List(int page = 1, int itemsPage = 10);
    }
}